models/listings.py

```diff
@@ -0,0 +1,85 @@
+import uuid
+from typing import Dict
+
+from lancedb.embeddings import get_registry
+from lancedb.pydantic import LanceModel, Vector
+from pydantic import AliasGenerator, ConfigDict, Field, field_validator, model_validator
+from typing_extensions import Self
+
+_openapi = get_registry().get("openai").create()
+
+
+class Listing(LanceModel):
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(
+            validation_alias=lambda name: name.lower().replace(" ", "_"),
+            serialization_alias=lambda name: " ".join(
+                word.capitalize() for word in name.split("_")
+            ),
+        )
+    )
+    id: str = Field(default_factory=lambda: uuid.uuid4().hex)
+    vector: Vector(_openapi.ndims()) = None  # type: ignore
+    image_vector: Vector(512) = None  # type: ignore
+    image: bytes = None
+    neighborhood: str
+    price: float
+    bedrooms: int
+    bathrooms: int
+    house_size: float
+    description: str
+    neighborhood_description: str
+    listing_summary: str = None
+
+    @field_validator("house_size", mode="before")
+    def parse_house_size(cls, value):
+        if value and isinstance(value, str):
+            return float(value.replace("sqft", "").replace(",", "").strip())
+        return value
+
+    @field_validator("price", mode="before")
+    def parse_price(cls, value):
+        if value and isinstance(value, str):
+            return float(value.replace("$", "").replace(",", ""))
+        return value
+
+    @model_validator(mode="after")
+    def validate_model(self) -> Self:
+        if not self.listing_summary:
+            self.listing_summary = get_listing_summary(self)
+
+
+def get_listing_summary(data: Dict | Listing) -> str:
```

Comment:
using `get` could return `None` or `False` if the key value was willingly set to `None` or `False`. In that case I always want the value to be `''` instead of `False` or `None`